tests/test_default_router.py

```diff
@@ -159,6 +159,10 @@ def test_invalid_field_name(self, template):
             ValueError,
             self.router.add_route, template, {}, ResourceWithId(-1))
 
+    def test_single_character_field_name(self):
+        self.router.add_route(
+            '/search?q={q}', {}, ResourceWithId(22))
```

Comment:
Ahh! Sorry, what I meant was that I think we need to add an assertion that verifies that the route actually works with the single character, not just that it doesn't raise an exception. So something like:
```
self.router.add_route('/item/{q}', {}, ResourceWithId(22))
__, __, params, __ = self.router.find('/item/1234')
self.assertEqual(params, {'q': '1234'})
```
This should make sure the routing actually works with your change.